msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SignInTest.kt

```diff
@@ -43,11 +114,59 @@ class SignInTest : NativeAuthPublicClientApplicationAbstractTest() {
         Assert.assertTrue((result as SignInError).isInvalidCredentials())
     }
 
+    /**
+     * Use email and OTP to get token and sign in (hero scenario 6, use case 2.2.1) - Test case 30
+     */
     @Test
-    fun testSignInSuccessSimple() = runTest {
-        val username = NativeAuthCredentialHelper.nativeAuthSignInUsername
-        val password = getSafePassword()
-        val result = application.signIn(username, password.toCharArray())
-        Assert.assertTrue(result is SignInResult.Complete)
+    fun testSuccessEmailOTP() = runTest {
+        var signInResult: SignInResult
+        var otp: String
+
+        retryOperation {
+            runBlocking {
+                val user = tempEmailApi.generateRandomEmailAddress()
+                signInResult = application.signIn(user)
+                Assert.assertTrue(signInResult is SignInResult.CodeRequired)
+                otp = tempEmailApi.retrieveCodeFromInbox(user)
+                val submitCodeResult = (signInResult as SignInResult.CodeRequired).nextState.submitCode(otp)
+                Assert.assertTrue(submitCodeResult is SignInResult.Complete)
+                "Success"
```

Comment:
Is there a reason for having this string here?

Reply:
Removed.